airflow/upgrade/checker.py

```diff
@@ -41,6 +41,15 @@ def check_upgrade(formatter, rules):
     return all_rule_statuses
 
 
+def list_checks():
+    print()
+    print("Upgrade Checks:")
+    for rule in ALL_RULES:
+        rule_name = rule.__class__.__name__
+        print("- {} [{}]".format(rule.title, rule_name))
```

Comment:
```suggestion
        print("- {} - {}".format(rule_name , rule.title))
```
Now when we have ordering, should we consider changing order of those two so the list really look like sorted?

Reply:
Makes sense! How does this look? (tweaked the formatting just a bit)
```
Upgrade Checks:
- VersionCheckRule: Check for latest versions of apache-airflow and checker
- AirflowMacroPluginRemovedRule: Remove airflow.AirflowMacroPlugin class
- BaseOperatorMetaclassRule: Ensure users are not using custom metaclasses in custom operators
- ChainBetweenDAGAndOperatorNotAllowedRule: Chain between DAG and operator not allowed.
- ConnTypeIsNotNullableRule: Connection.conn_type is not nullable
- DbApiRule: Hooks that run DB functions must inherit from DBApiHook
- FernetEnabledRule: Fernet is enabled by default
- GCPServiceAccountKeyRule: GCP service account key deprecation
- HostnameCallable: Unify hostname_callable option in core section
- ImportChangesRule: Changes in import paths of hooks, operators, sensors and others
- LegacyUIDeprecated: Legacy UI is deprecated by default
- LoggingConfigurationRule: Logging configuration has been moved to new section
- MesosExecutorRemovedRule: Removal of Mesos Executor
- NoAdditionalArgsInOperatorsRule: No additional argument allowed in BaseOperator.
- PodTemplateFileRule: Users must set a kubernetes.pod_template_file value
- SendGridEmailerMovedRule: SendGrid email uses old airflow.contrib module
- TaskHandlersMovedRule: Changes in import path of remote task handlers
- UndefinedJinjaVariablesRule: Jinja Template Variables cannot be undefined
- UniqueConnIdRule: Connection.conn_id is not unique
```

Reply:
Would be awesome to have a tabular format (makes it more readable) like below but not a requirement for this PR :)

```
| ClassName                                | Description                                                       |
|------------------------------------------|-------------------------------------------------------------------|
| VersionCheckRule                         | Check for latest versions of apache-airflow and checker           |
| AirflowMacroPluginRemovedRule            | Remove airflow.AirflowMacroPlugin class                           |
| BaseOperatorMetaclassRule                | Ensure users are not using custom metaclasses in custom operators |
| ChainBetweenDAGAndOperatorNotAllowedRule | Chain between DAG and operator not allowed.                       |
| ConnTypeIsNotNullableRule                | Connection.conn_type is not nullable                              |
| DbApiRule                                | Hooks that run DB functions must inherit from DBApiHook           |
| FernetEnabledRule                        | Fernet is enabled by default                                      |
| GCPServiceAccountKeyRule                 | GCP service account key deprecation                               |
| HostnameCallable                         | Unify hostname_callable option in core section                    |
| ImportChangesRule                        | Changes in import paths of hooks, operators, sensors and others   |
| LegacyUIDeprecated                       | Legacy UI is deprecated by default                                |
| LoggingConfigurationRule                 | Logging configuration has been moved to new section               |
| MesosExecutorRemovedRule                 | Removal of Mesos Executor                                         |
| NoAdditionalArgsInOperatorsRule          | No additional argument allowed in BaseOperator.                   |
| PodTemplateFileRule                      | Users must set a kubernetes.pod_template_file value               |
| SendGridEmailerMovedRule                 | SendGrid email uses old airflow.contrib module                    |
| TaskHandlersMovedRule                    | Changes in import path of remote task handlers                    |
| UndefinedJinjaVariablesRule              | Jinja Template Variables cannot be undefined                      |
| UniqueConnIdRule                         | Connection.conn_id is not unique                                  |
```